### PR TITLE
Test for Windows osquery daemon

### LINKED ISSUE #1993: Tests for duplicate instance abstraction

### LINKED ISSUE #2007: Runs as a Windows service

### LINKED ISSUE #2008: Tests for Windows service functionality

### PR DESCRIPTION
**Pull request by yying**
Windows specific tests for `osqueryd`

Closes #1993 
Closes #2008

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by theopolis** [comment]
ok to test

**Comment by osqueryer** [comment]
Build 9a004bd6d2007e6aa131cb8aa97a83d5c74057b1 (Job: 2958) failed :-1:

File: tools/tests/test_osqueryd_win.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        sys.exit(-1)
+    try:
+        os.listdir("\\Windows\\Temp")
+    except WindowsError:
+        sys.exit(-1)
+
+
+def sc(*args):
+    try:
+        output = subprocess.check_output(["sc.exe"] + list(args))
```
Comment (PoppySeedPlehzr) [inline tools/tests/test_osqueryd_win.py]: Any chance you could grab stdout and stderr here and print them in the exception? Either that or grab the exception and display what it its? Might be nice for debugging if something starts going wrong?

**Comment by osqueryer** [comment]
Build c458ca55cd459cdf3b6002163fe6707dea37e989 (Job: 2960) failed :-1:

**Pull request merged**